enhancements/network/cloud-egress-ip.md

```diff
@@ -0,0 +1,652 @@
+maximum across the VPC which is variable to each project. The OpenShift clusters
+that have been studied have shown this value to be 15.000 defined for the
+entire VPC.
+
+[the GCP API documentation]:
+https://cloud.google.com/sdk/gcloud/reference/compute/instances/network-interfaces/update
+
+[GCP limits]: https://cloud.google.com/vpc/docs/quota#per_instance
+
+#### Model
+
+This cloud component is not intended on being "user facing", its purpose is a
+means to an end for implementing a OpenShift networking feature in the cloud (in
+this case: egress IP) by the networking stack. As such the primary "user" of
+this cloud component will be the network plugins themselves. This impacts the
+design decision with regards to the cloud component's failure notification, see
+[CRD section](#crd).
+
+##### network plugin <-> cloud API controller
+
+The network plugin needs to remain "source of truth" of the egress IP setup. The
+reason for this is that the network plugin is the one and only component which
+health checks routes to egress nodes and re-balances the egress IP assignment
+depending on the result. This component will need to execute what the network
+plugin has determined is best. This is done since there is no traffic load
+balancing done today for egress IP intended traffic. The network plugin
+constantly initiates a TCP connection to each egress node and depending on the
+success/failure of that: re-balances the egress assignment to other nodes (thus
+avoiding packets being sent to a egress node which is currently un-reachable).
+Going forward, both network plugins will come to support traffic load-balancing
+and a more sophisticated route health detection. That means that the underlying
+networking solution (OVS in the case of both OVN-Kubernetes and openshift-sdn)
+will manage the route health check. It is worth noting that the "amount of
+re-balancing" done by the network plugin is expected to be reduced, since the
+network fabric will be able to load balance traffic between all egress nodes
+provided and always use a healthy node (thus not requiring the network plugin to
+perform an egress IP re-assignment every time a route goes down).
+
+Given the above, it is however important that the network plugin is conveyed the
+cloud's information concerning the nodes' subnet range. This is especially
+evident on GCP, as without it: the network plugin won't be able to perform any
+assignments at all, given the nodes' default `/32` subnet.
+
+#### Inter-component communication
+
+Multiple pieces of information needs to be communicated between the network
+plugins and this cloud component:
+
+1. The cloud's node subnet and capacity information.
+2. The egress IP assignment to be performed by the cloud component
+3. The status of the egress IP assignment performed by the cloud component
+
+Two models have been decided for this: a CRD based approach and a node
+annotated approach.
```

Comment:
This sounds like you're going to present two different models below which we haven't yet decided between, but then you just explain how annotations will be used for 1 and a CRD for 2 and 3.